packages/block-editor/src/components/use-on-block-drop/test/index.js

```diff
@@ -0,0 +1,413 @@
+			targetBlockIndex,
+			getBlockIndex,
+			getClientIdsOfDescendants,
+			moveBlocksToPosition
+		);
+		eventHandler( event );
+
+		expect( moveBlocksToPosition ).not.toHaveBeenCalled();
+	} );
+
+	it( 'inserts blocks if the drop is valid', () => {
+		const sourceClientIds = [ '5' ];
+		const sourceRootClientId = '0';
+		const targetRootClientId = '1';
+		const targetBlockIndex = 0;
+		const getBlockIndex = jest.fn( () => 1 );
+		const getClientIdsOfDescendants = () => [];
+		const moveBlocksToPosition = jest.fn();
+
+		const event = {
+			dataTransfer: {
+				getData() {
+					return JSON.stringify( {
+						type: 'block',
+						srcRootClientId: sourceRootClientId,
+						srcClientIds: sourceClientIds,
+					} );
+				},
+			},
+		};
+
+		const eventHandler = onBlockDrop(
+			targetRootClientId,
+			targetBlockIndex,
+			getBlockIndex,
+			getClientIdsOfDescendants,
+			moveBlocksToPosition
+		);
+		eventHandler( event );
+
+		expect( moveBlocksToPosition ).toHaveBeenCalledWith(
+			sourceClientIds,
+			sourceRootClientId,
+			targetRootClientId,
+			targetBlockIndex
+		);
+	} );
+
+	it( 'adjusts the index blocks are dropped at when moved down under the same root block', () => {
+		const sourceClientIds = [ '5', '6', '7' ];
+		const sourceRootClientId = '0';
+		const targetRootClientId = sourceRootClientId;
+		const targetBlockIndex = 5;
+		const getBlockIndex = jest.fn( () => 1 );
+		// Dragged block is being dropped as a descendant of itself.
+		const getClientIdsOfDescendants = () => [];
+		const moveBlocksToPosition = jest.fn();
+
+		const event = {
+			dataTransfer: {
+				getData() {
+					return JSON.stringify( {
+						type: 'block',
+						srcRootClientId: sourceRootClientId,
+						// The dragged block is being dropped as a child of itself.
+						srcClientIds: sourceClientIds,
+					} );
+				},
+			},
+		};
+
+		const insertIndex = targetBlockIndex - sourceClientIds.length;
+
+		const eventHandler = onBlockDrop(
+			targetRootClientId,
+			targetBlockIndex,
+			getBlockIndex,
+			getClientIdsOfDescendants,
+			moveBlocksToPosition
+		);
+		eventHandler( event );
+
+		expect( moveBlocksToPosition ).toHaveBeenCalledWith(
+			sourceClientIds,
+			sourceRootClientId,
+			targetRootClientId,
+			insertIndex
+		);
+	} );
+} );
+
+describe( 'onFileDrop', () => {
+	it( 'does nothing if hasUploadPermissions is false', () => {
+		const updateBlockAttributes = jest.fn();
+		const insertBlocks = jest.fn();
+		const targetRootClientId = '1';
+		const targetBlockIndex = 0;
+		const uploadPermissions = false;
+
+		const onFileDropHandler = onFileDrop(
+			targetRootClientId,
+			targetBlockIndex,
+			uploadPermissions,
+			updateBlockAttributes,
+			insertBlocks
+		);
+		onFileDropHandler();
+
+		expect( findTransform ).not.toHaveBeenCalled();
+		expect( insertBlocks ).not.toHaveBeenCalled();
+	} );
+
+	it( 'does nothing if the block has no matching file transforms', () => {
+		findTransform.mockImplementation( () => {} );
```

Comment:
Hmm, I suppose not strictly required since `jest.fn()` also returns nothing, though I feel like having this line helps demonstrate that it's performing the inverse of the following test.

I switched it over to use `noop` and also added some comments.